tests/README.md

```diff
@@ -0,0 +1,38 @@
+## Test Organization
+This folder contains all the tests for the Brevitas repository.
+
+Everytime a new function is added or modified, it is highly recommeded to adapt the associated test cases if present, or to add new ones if tests are missing.
+
+We are in the process of expanding and re-organising the test suite, thus information and descriptions contained here might change in the future.
+
+Brevitas test suite is built around the following packages:
+- Pytest
+- Hypothesis
+- Pytest Cases
+- Mock (Pytest Mock)
+
+Please feel free to familiriasize with their basic usage before contributing to new tests on Brevitas.
+
+### Folder Structure
+The root tests folder structure is based on requirements:
+- brevitas: tests the main functionalities of the library. Only brevitas and the test packages should be needed to run these tests.
+- brevitas_examples: tests for the quantization examples in _src\_brevitas\_examples_.
+- brevitas_finn: tests for the integration with the finn export-flow.
+- brevitas_ort: tests for the different brevitas onnx exports, leveraging onnxruntime.
+
+In particular, within the _brevitas_ folder, the subfolder structure matches the one in _src\_brevitas_.
+A class in _src\_brevitas\_core_ will have its corresponding test in _tests\_brevitas\_core_.
+
+### Adding New Tests
+When adding a new test, make sure to respect the folder structure described above.
+When adding or modifying a file in _src\_brevitas_,  make sure that the corresponding test is adapted (or added, if missing) in the respective test folder. 
+
+
+Generally speaking, there are two main types of tests used for Brevitas. 
```

Comment:
Tests with hypothesis can also leverage fixtures, it's not one or another. Hypothesis is an extra layer on top of things when we want to stress the numerics.